util/movement-types/src/lib.rs

```diff
@@ -50,25 +60,52 @@ impl Transaction {
 		Self { data, sequence_number, id }
 	}
 
-	pub fn id(&self) -> Id {
-		self.id.clone()
+	pub fn id(&self) -> &Id {
```

Comment:
This can return `Id` by value, to avoid a lot of inconvenience at the call sites. The value is supposed to be small and `Copy` as suggested in another comment.

To benefit from optimizations that should largely make these distinctions irrelevant, put an `#[inline]` attribute on the method. This will make the implementation visible outside of the crate, the compiler won't do it for non-generic types otherwise and I'm not sure LTO can make up for this in all cases.

Reply:
Oh, really? I wasn't aware that the compiler needed so much help inlining owned accessors of this sort. Can you link me to something for my own knowledge?

Reply:
It's more about what metadata is available in the compiled crates. `#[inline]` forces the compiler to provide the body for inlining across crates (at some cost to compile time), otherwise the non-generic method is just linked as a symbol. There may be also LTO tricks nowadays to make this unnecessary.